### ISSUE TITLE
Replace `binary-install`

### ISSUE THREAD

**Issue by ebadta81** [akesson/cargo-leptos#24]
Hello Henrik,

when I do ```cargo leptos build -r``` I got:

```
    Finished release [optimized] target(s) in 18.61s
       Cargo finished cargo build --no-default-features --features=hydrate --lib --target=wasm32-unknown-unknown --release
Error: at `src/main.rs@147:33`

Caused by:
    0: at `src/main.rs@163:36`
    1: at `src/run/wasm.rs@20:34`
    2: at `src/run/wasm.rs@52:41`
    3: at `src/run/wasm.rs@77:35`
    4: Unable to download wasm-opt for macos aarch64 due to: failed to extract tarball from https://github.com/WebAssembly/binaryen/releases/download/version_111/binaryen-version_111-arm64-macos.tar.gz
```

I installed the latest cargo-leptos, and without ```-r```, the build woks.

**Comment by akesson** [comment]
Ok, I can reproduce it. I'm using the `binary-install` crate for it and it works really poorly. I'll need to replace it. I'll change the title to reflect that.

As a workaround do `brew install binaryen` and make sure that `wasm-opt` runs on the command line.

**Comment by akesson** [comment]
Related to #21

**Comment by michalvavra** [comment]
I would suggest to look at something that is using `reqwest` instead of `curl` (like in `binary-install`.

----

TLDR;

After a couple of hours of debugging, I found out that the underlying problem is `curl` crate (used by `binary-install`).

I’ve tried many things on Windows:

- Updating curl version (https://github.com/curl/curl/issues/4427)
- Installing Perl (https://github.com/cloudflare/wrangler/issues/1765 and )
- Using `curl` native features (`static-ssl` and `static-curl`)

I wasn’t able to download a binary from Windows. Directly via `curl` command or `curl` crate.
It’s something between the `curl` and Windows that is the problem.

`reqwest` downloads GitHub binaries without problem for me.

**Comment by akesson** [comment]
Oh, fantastic! A great help, thanks a lot!!!

**Comment by michalvavra** [comment]
I can take over this one. I haven't found any `binary-install` alternative.
But I have now working prototype on Windows (using `reqwest`).

It needs a bit of polishing and testing but I can deliver the PR over the weekend.

**Comment by akesson** [comment]
Great. Go ahead!
